src/components/org/apache/jmeter/extractor/json/jsonpath/JSONPostProcessor.java

```diff
@@ -0,0 +1,236 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.jmeter.extractor.json.jsonpath;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.jmeter.processor.PostProcessor;
+import org.apache.jmeter.samplers.SampleResult;
+import org.apache.jmeter.testelement.AbstractScopedTestElement;
+import org.apache.jmeter.testelement.ThreadListener;
+import org.apache.jmeter.threads.JMeterContext;
+import org.apache.jmeter.threads.JMeterVariables;
+import org.apache.jmeter.util.JMeterUtils;
+import org.apache.jorphan.logging.LoggingManager;
+import org.apache.jorphan.util.JOrphanUtils;
+import org.apache.log.Logger;
+
+/**
+ * JSON-PATH based extractor
+ * @since 2.14
+ */
+public class JSONPostProcessor extends AbstractScopedTestElement implements Serializable, PostProcessor, ThreadListener{
+
+    private static final long serialVersionUID = 1320798545214331506L;
+    private static final Logger log = LoggingManager.getLoggerForClass();
+    private static final String JSON_PATH_EXPRESSIONS = "ULP.json.jsonPathExprs"; // $NON-NLS-1$
+    private static final String REFERENCE_NAMES = "ULP.json.referenceNames"; // $NON-NLS-1$
+    private static final String DEFAULT_VALUES = "ULP.json.defaultValues"; // $NON-NLS-1$
+    private static final String MATCH_NUMBERS = "ULP.json.match_numbers"; // $NON-NLS-1$
+    private static final String COMPUTE_CONCATENATION = "ULP.json.compute_concat"; // $NON-NLS-1$
+    private static final String REF_MATCH_NR = "_matchNr"; // $NON-NLS-1$
+    private static final String ALL_SUFFIX = "_ALL"; // $NON-NLS-1$
+    private static final String SEPARATOR = ";"; // $NON-NLS-1$
+    public static final boolean COMPUTE_CONCATENATION_DEFAULT_VALUE = false;
+    
+    private static final ThreadLocal<JSONManager> localMatcher = new ThreadLocal<JSONManager>() {
+        @Override
+        protected JSONManager initialValue() {
+            return new JSONManager();
+        }
+    };
+    public void process() {
+        JMeterContext context = getThreadContext();
+        JMeterVariables vars = context.getVariables();
+        String jsonResponse = "";
+        if (isScopeVariable()){ 
+            jsonResponse = vars.get(getVariableName());
+            if(log.isDebugEnabled()) {
+                log.debug("JSON Extractor is using variable:"+getVariableName()+" which content is:"+jsonResponse);
+            }
+        } else {
+            SampleResult previousResult = context.getPreviousResult();
+            if (previousResult == null) {
+                return;
+            }
+            jsonResponse = previousResult.getResponseDataAsString();
+            if(log.isDebugEnabled()) {
+                log.debug("JSON Extractor processing result:"+jsonResponse);
```

Comment:
jsonResponse is the the content on which the extraction should take place not the processing result, right?